### PR TITLE
Count what a local response norm and an Lp pool compute

### PR DESCRIPTION
**Pull request by raimbekovm**
## Summary

`nn.LocalResponseNorm`, `nn.CrossMapLRN2d` and the `nn.LPPool1d/2d/3d` family reached no rule, so each reported 0.0 MACs while `report_missing=True` named it. Both do real per-element work over a window, and `count_normalization` is the wrong shape for them: it is `2 * numel` and reads no window at all.

`count_lrn` charges one square per element and the `size` window adds, then the window's divide, the alpha multiply, the k add, the beta power and the final divide. torch zero-pads the channel axis by `size // 2` and `(size - 1) // 2` and runs a stride-1 pool over it, so every window is exactly `size` wide and the total is `numel * (size + 6)`. `nn.CrossMapLRN2d` is the same arithmetic over the same four arguments and shares the rule.

`count_lp_pool` reaches all three classes through the `_LPPoolNd` base they share, the way the padding, dropout and unpool families already do. It charges one power per input and one add per input its windows read, then the pool's divide, the multiply by the sign, the multiply by the kernel area and the p-th root for every output. `nn.LPPool` never pads, which is why its window loop has no padding term and does not go through `count_avgpool`. At `norm_type = ±inf` torch dispatches to `max_pool` over absolute values, so the rule returns early and reports zero, as `nn.MaxPool2d` already does.

`nn.FractionalMaxPool2d` and `nn.FractionalMaxPool3d` already resolve to `zero_ops` and are untouched.

## Validation

- `nn.LocalResponseNorm(5)` over a `(1, 8, 16, 16)` input reports 22528 where it reported 0.0, and `nn.LPPool2d(2, 2)` over the same input reports 6144. Verified through both `profile` and `profile_origin`.
- Every expected count is written as its own enumeration rather than as a literal. The LRN figure is reached twice: once from the seven terms above and once by handing the same zero-padded stride-1 window to the shipped `count_avgpool`, which agree at `size` 2, 3 and 5.
- Covered: `stride=None`, mixed tuple kernels and strides, `ceil_mode=True` windows clipped by the input bound, unbatched inputs, rank-3 inputs, fractional and negative `norm_type`, and a `norm_type` torch coerces from a string.
- Eight `nn.AvgPool` configurations are pinned as (window contributions, output elements) and unchanged, since the LRN cross-check leans on that rule.
- Neither rule belongs in the `stride=` estimator's `fixed_ops`: both costs are proportional to the input element count. A conv model ending in either reports the same total directly and through `stride=32` for non-overlapping windows; the overlapping case is 0.6% low, which is the clipped last window and not this change — `nn.AvgPool2d(3, stride=2)` with the same geometry is already 0.5% low on current main.
- A conv, batch-norm, ReLU and adaptive-average-pool model reports 4612.0 before and after, and the shipped tests pass.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds THOP operation-counting support for local response normalization and Lp pooling layers.

### 📊 Key Changes
- Registers `nn.LocalResponseNorm` and legacy `nn.CrossMapLRN2d` with a new LRN counting hook.
- Registers the `_LPPoolNd` family with a new Lp pooling counting hook covering 1D, 2D, and 3D variants.
- Accounts for pooling windows, norm type behavior, and per-element normalization or pooling operations.

### 🎯 Purpose & Impact
- Improves FLOP/operation estimates for models using LRN and Lp pooling layers.
- Supports accurate profiling across current and legacy PyTorch layer names.
- Lp pooling with infinite norm is treated as a selection-based max-pooling path and contributes no counted arithmetic operations.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/thop` 🚀 PR! This automated message confirms receipt of your changes; an engineer will assist with the review. To ensure a seamless integration of your LRN and Lp pooling operation-counting work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/thop/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/thop` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The Lp-pooling and LocalResponseNorm hooks follow the existing hook structure and account for the relevant window work. One compatibility issue needs fixing: the unconditional legacy CrossMapLRN2d reference breaks imports on current PyTorch.

💬 Posted 1 inline comment

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The new hooks are registered through the existing MRO-based rule resolution, and the operation counts account for LRN channel windows and Lp pooling dimensions, strides, partial ceil-mode windows, and infinity norms without affecting existing rules.

**Comment by glenn-jocher** [comment]
Closing this one on the duplication and the constants.

**1. `count_lp_pool` re-implements `count_avgpool`'s window loop.** Side by side on this branch:

```
basic_hooks.py:123  windows *= sum(max(min(i*s - p + k, upper) - max(i*s - p, lower), 0) for i in range(output))  # count_avgpool
basic_hooks.py:136  windows *= sum(min(i*s + k, size) - i*s for i in range(output))                              # count_lp_pool
```

The second is the first with the padding terms dropped. AGENTS.md principle 2 asks for consolidation in the shared owner, and the PR body argues *for* the fork ("`nn.LPPool` never pads, which is why ... it does not go through `count_avgpool`") — but never padding means `p = 0`, which the existing owner already handles.

**2. The per-output constants are not derivable.** `+ 4 * y.numel()` and `(m.size + 6)` bundle a divide, two multiplies and a root into hand-tuned integers. The LRN cross-check against `count_avgpool` validates the *window* term, not the `+ 6`.

**3. `abs(float(m.norm_type)) == float("inf")` with "torch accepts a string norm"** is impossible-state handling — a string `norm_type` fails inside `F.lp_pool2d` itself, so the coercion guards a state that cannot reach the hook.

**4. No consumer.** `nn.LocalResponseNorm` is AlexNet-era and superseded by BatchNorm, `nn.CrossMapLRN2d` is a legacy shim, and none of the LRN or LpPool families appear in `ultralytics/nn/`.

27 lines of duplicated, hand-tuned arithmetic for two dead layer families is the wrong trade.
